struct.md

```diff
@@ -0,0 +1,337 @@
+As of now it does not state how exporters are affected.
+
+## Rationale and Recommendations on intended usage
+
+### Background
+
+VSS currently supports only the following types:
+
+* Integer-based types (e.g. uint8, int32)
+* Float-based types (float, double)
+* String
+* Boolean
+
+In addition to this VSS supports arrays of the types given above. There are cases where this may not be sufficient.
+Typical use-cases are when something cannot be described by a single value, but multiple values are needed.
+
+A few hypothetical examples include:
+
+* GPS locations, where latitude and longitude must be handled together
+* Obstacles - where each obstacle may contain information like, category, probability and location
+* Errors/Warnings - where each item might contain information on category and priority
+
+### Intended usage
+
+The proposed struct support in VSS is introduced to facilitate logical binding/grouping of data that originates from the same source.
+It is intended to be used only when it is important that the data is read or written in an atomic operation.
+It is not intended to be used to specify how data shall be packaged and serialized when transported.
+
+By this reason VSS-project will not introduce smaller datatypes (like `uint1`,`uint4`) to enable bit-encoding of data.
+The order of elements in a struct is from a VSS perspective considered as arbitrary.
+The VSS-project will by this reason not publish guidelines on how to order items in the struct to minimize size,
+and no concept for introducing padding will exist.
+
+Structs shall be used in VSS standard catalog only when considered to give a significant advantage compared to using only primitive types.
+
+
+### Structs vs. Aggregate
+
+VSS supports a keyword `aggregate` that can be used on branches to indicate that the branch preferably shall be read and written in atomic operations.
+The keyword is however currently not used in the standard catalog, and it is not known if any implementation exists that actually consider it.
+There have been criticism that `aggregate` changes the semantic meaning of branches and signals, i.e. that a signal is no longer handed as an independent object.
+The exact meaning of `aggregate` is furthermore not well defined by VSS. Shall for example a write request (or update of sensor values) be rejected by an implementation
+if not all signals in the branch are updated in the same operation.
+Semantic interpretation is also ambiguous if the branch contains a mix of sensors, attributes and actuators.
+
+Using structs as datatype is better aligned with the view that VSS signals are independent objects, and the semantic ambiguities related to `aggregate` are not present for structs.
+
+**TBD**:
+
+*Shall we keep `aggregate`, or remove it? If it is to be kept, does it need to be better defined to be able to argue why it is still needed? Do we know of anyone or any implementation actually using it?*
```

Comment:
I have some difficulties fining a good argument why `aggregate`shall be kept, and if kept what it actually should mean.

Reply:
Aggregate: To be used when a collection of sensors need to be treated as a unit (Existing definition)
Struct: To be used when a single sensor output is non-primitive/struct/higher-order/complex.

In essence, aggregate is used for grouping sensors just as today while struct is used for defining a single sensor.

If you want to deprecated aggregate, that means its existing utility is also not justifiable?

Reply:
Copying the comment from @UlfBj here:

_If it is to be kept then I think a credible example shall be presented._

Reply:
We have previously said that `Vehicle.CurrentLocation` could be a branch were `aggregate=true` could make sense. But one could then also argue that it is the same sensor that sets/publish lat/long/altitude, so that they would qualify for the criteria for being in a struct as well.